gammapy/estimators/map/excess.py

```diff
@@ -67,9 +72,17 @@ def convolved_map_dataset_counts_statistics(dataset, kernel, mask, correlate_off
             with np.errstate(invalid="ignore", divide="ignore"):
                 alpha = acceptance_on_convolve / acceptance_off
 
-        return WStatCountsStatistic(
-            n_on_conv.data, n_off.data, alpha.data, npred_sig_convolve.data
-        )
+        if full_output:
+            maps = Maps(
+                acceptance_on=acceptance_on, acceptance_off=acceptance_off, alpha=alpha
+            )
+            return WStatCountsStatistic(
+                n_on_conv.data, n_off.data, alpha.data, npred_sig_convolve.data
+            ), maps
+        else:
+            return WStatCountsStatistic(
+                n_on_conv.data, n_off.data, alpha.data, npred_sig_convolve.data
+            )
```

Comment:
maybe introduce a private function get_convolved_maps(dataset, kernel, mask, correlate_off) that returns a dict of `convolved_maps` and use it in `convolved_map_dataset_counts_statistics(convolved_maps, staty_type=dataset.stat_type)` (it is not part of the public api so it think it is fine to change its arguments).
Then in `estimate_excess_map` you add `if alpha in selection_optional : maps['alpha']=...`